src/rgw/rgw_pubsub.h

```diff
@@ -432,7 +432,7 @@ struct rgw_pubsub_topic {
       // fill in the persistent queue's location if it wasn't present.
       // rgw_pubsub_dest::decode() couldn't handle this itself because of the
       // dependency on rgw_pubsub_topic::name
-      if (dest.persistent && dest.persistent_queue.empty()) {
+      if (dest.persistent_queue.empty()) {
```

Comment:
the earlier check makes more sense as `persistent_queue` is applicable only for persistent notification.
this change will now add name for `non-persistent` notification as well

Reply:
hmm, CreateTopic and SetTopicAttributes don't actually know whether or not it previously set, so i don't think it's safe to change the value of persistent_queue at that point

Reply:
> hmm, CreateTopic and SetTopicAttributes don't actually know whether or not it previously set, so i don't think it's safe to change the value of persistent_queue at that point

> CreateTopic and SetTopicAttributes don't actually know whether or not it previously set
why? while reading the topic from rados, it will know whether the topic was persistent or not.

Reply:
oops, you're right. these ops both try to load the existing topic first